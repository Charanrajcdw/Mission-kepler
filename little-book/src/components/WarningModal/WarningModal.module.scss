.warningModalContainer {
  position: absolute;
  display: flex;
  justify-content: center;
  align-items: center;
  top: 0;
  right: 0;
  left: 0;
  bottom: 0;
  background-color: rgba(0, 0, 0, 0.1);
  z-index: 99;

  .warningModalContent {
    width: 35%;
    height: 20vh;
    padding: 2.5rem;
    background-color: var(--modal-bg);

    .title {
      font-size: 2rem;
      font-weight: 500;
      text-align: center;
      margin-bottom: 28px;
    }

    .buttonsContainer {
      display: flex;
      justify-content: center;
      gap: 20px;
    }
  }
}
