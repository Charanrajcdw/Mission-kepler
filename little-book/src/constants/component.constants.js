export const MODAL = {
  members: "Members",
  newBlog: "Add New Blog",
  remove: "Remove",
};

export const MEMBERS_LIST = {
  title: "Members",
  noMembers: "No members available",
};

export const BLOG_FORM = {
  title: "Add New Blog",
  button: "ADD",
  titlePlaceholder: "Name your blog",
  detailsPlaceholder: "Write Content Here ..",
  invalidValues: "Invalid title or description",
  invalidBlog: "Blog title already exists",
  blogAdded: "New Blog added successfully",
  localBlog: "local",
};

export const WARNING_MODAL = {
  title: "Unsaved changes will be lost. Are you sure to continue?",
  continue: "Continue",
  cancel: "Cancel",
};
