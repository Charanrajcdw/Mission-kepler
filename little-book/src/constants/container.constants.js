export const SIDEBAR = {
  logo: "Little Book",
  filter: "FILTER",
  noFilter: "Add blogs to show filters",
  viewMembers: "View Members",
  switchToDark: "Switch to Dark Mode",
  switchToLight: "Switch to Light Mode",
  blogs: "Blogs",
};
