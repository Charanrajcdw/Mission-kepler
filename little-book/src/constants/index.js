export * from "./url.constants";
export * from "./theme.constants";
