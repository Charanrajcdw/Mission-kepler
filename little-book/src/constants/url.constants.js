export const API_URLS = {
  blogs: "https://jsonmockserver.vercel.app/api/blogs",
  users: "https://jsonmockserver.vercel.app/api/users",
  images: "https://jsonmockserver.vercel.app/",
};
