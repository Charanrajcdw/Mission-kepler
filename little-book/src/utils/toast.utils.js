import { toast } from "react-toastify";

export const showSuccessToast = (message, currentTheme) => {
  toast.success(message, {
    position: "top-right",
    autoClose: 2000,
    closeOnClick: true,
    pauseOnHover: true,
    draggable: true,
    theme: currentTheme,
  });
};
