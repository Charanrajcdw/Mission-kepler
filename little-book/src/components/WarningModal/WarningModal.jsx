import { useDispatch } from "react-redux";
import PropTypes from "prop-types";
import styles from "./WarningModal.module.scss";
import Button from "../Button/Button";
import { MODAL, WARNING_MODAL } from "../../constants";
import { blogActions } from "../../store";

const WarningModal = ({ modalHandler }) => {
  const dispatch = useDispatch();

  const continueHandler = () => {
    modalHandler(MODAL.remove, false);
    dispatch(blogActions.modifyEditStatus(false));
  };

  const cancelHandler = () => {
    modalHandler(false, false);
  };

  return (
    <div className={styles.warningModalContainer}>
      <div className={styles.warningModalContent}>
        <p className={styles.title}>{WARNING_MODAL.title}</p>
        <div className={styles.buttonsContainer}>
          <Button className="blueBtn" clickHandler={cancelHandler}>
            {WARNING_MODAL.cancel}
          </Button>
          <Button className="pinkBtn" clickHandler={continueHandler}>
            {WARNING_MODAL.continue}
          </Button>
        </div>
      </div>
    </div>
  );
};

WarningModal.propTypes = {
  modalHandler: PropTypes.func.isRequired,
};

export default WarningModal;
