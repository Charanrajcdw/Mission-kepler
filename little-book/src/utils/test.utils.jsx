import { configureStore } from "@reduxjs/toolkit";
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import ThemeContextProvider from "../contexts/theme.context";
import blogSlice from "../store/slices/blogSlice";
import { THEME } from "../constants";

const customRender = (
  ui,
  {
    preloadedState = {},
    store = configureStore({
      reducer: { blogs: blogSlice.reducer },
      preloadedState,
    }),
    theme = THEME.light,
    ...options
  } = {}
) => {
  const ProviderWrapper = ({ children }) => {
    return (
      <ThemeContextProvider theme={theme}>
        <Provider store={store}>{children}</Provider>
      </ThemeContextProvider>
    );
  };
  return render(ui, { wrapper: ProviderWrapper, ...options });
};

const customCreateSnapshot = (
  ui,
  {
    preloadedState = {},
    store = configureStore({
      reducer: { blogs: blogSlice.reducer },
      preloadedState,
    }),
    theme = THEME.light,
  } = {}
) => {
  const ProviderWrapper = ({ children }) => {
    return (
      <ThemeContextProvider theme={theme}>
        <Provider store={store}>{children}</Provider>
      </ThemeContextProvider>
    );
  };
  return renderer.create(<ProviderWrapper>{ui}</ProviderWrapper>).toJSON();
};

export * from "@testing-library/react";
export { default as userEvent } from "@testing-library/user-event";
export { customRender as render, customCreateSnapshot as createSnapshot };
