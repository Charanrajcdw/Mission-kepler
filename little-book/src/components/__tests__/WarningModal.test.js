import { userEvent, render, screen, createSnapshot } from "../../utils/test.utils.jsx";
import WarningModal from "../WarningModal/WarningModal";
import { MODAL, WARNING_MODAL } from "../../constants";

describe("Modal", () => {
  it("should render Warning Modal", () => {
    render(<WarningModal modalHandler={() => {}} />);
    const modalElement = screen.getByText(WARNING_MODAL.title);
    const cancelButton = screen.getByText(WARNING_MODAL.cancel);
    const continueButton = screen.getByText(WARNING_MODAL.continue);
    expect(modalElement).toBeInTheDocument();
    expect(cancelButton).toBeInTheDocument();
    expect(continueButton).toBeInTheDocument();
  });

  it("should match modal snapshot", () => {
    const warningModal = createSnapshot(<WarningModal modalHandler={() => {}} />);
    expect(warningModal).toMatchSnapshot();
  });

  it("should call modal close handler", async () => {
    const user = userEvent.setup();
    const modalHandler = jest.fn();
    render(<WarningModal modalHandler={modalHandler} />);
    const cancelButton = screen.getByText(WARNING_MODAL.cancel);
    await user.click(cancelButton);
    expect(modalHandler).toHaveBeenCalledWith(false, false);
  });

  it("should call modal continue handler", async () => {
    const user = userEvent.setup();
    const modalHandler = jest.fn();
    render(<WarningModal modalHandler={modalHandler} />);
    const continueButton = screen.getByText(WARNING_MODAL.continue);
    await user.click(continueButton);
    expect(modalHandler).toHaveBeenCalledWith(MODAL.remove, false);
  });
});
