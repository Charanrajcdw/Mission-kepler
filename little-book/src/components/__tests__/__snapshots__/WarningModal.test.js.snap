// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should match modal snapshot 1`] = `
<div
  className="warningModalContainer"
>
  <div
    className="warningModalContent"
  >
    <p
      className="title"
    >
      Unsaved changes will be lost. Are you sure to continue?
    </p>
    <div
      className="buttonsContainer"
    >
      <button
        className="btn blueBtn"
        onClick={[Function]}
      >
        Cancel
      </button>
      <button
        className="btn pinkBtn"
        onClick={[Function]}
      >
        Continue
      </button>
    </div>
  </div>
</div>
`;
